src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -233,9 +241,20 @@ private void OnDescriptionChanged(object sender, ServerDescriptionChangedEventAr
                 try { handler(this, e); }
                 catch { } // ignore exceptions
             }
+
+        }
+
+        private void OnMonitorDescriptionChanged(object sender, ServerDescriptionChangedEventArgs e)
+        {
+            var currentDescription = Interlocked.CompareExchange(ref _currentDescription, value: null, comparand: null);
+            if (e.NewServerDescription.HeartbeatException != null ||
+                currentDescription.TopologyVersion.IsStalerThanServerResponse(e.NewServerDescription.TopologyVersion))
```

Comment:
See [this part of the spec commit](https://github.com/mongodb/specifications/commit/7dd9c008be6edaf65368787d5e91c2ac3aeb95b8#diff-62b4da2e38e7d2a9c1a2d92d315895bbR1640-R1645)
```
 def isStaleServerDescription(topologyDescription, server):
        # True if the new ServerDescription's topologyVersion is greater than
        # or equal to the current server's.
        currentServer = topologyDescription.servers[server.address]
        currentTopologyVersion = currentServer.topologyVersion
        return compareTopologyVersion(currentTopologyVersion, server.topologyVersion) > 0
```

Reply:
I am open to the idea of adding an `isStaleServerDescription` method for an easier mental model between the spec and our code, but the currently implemented logic is equivalent.